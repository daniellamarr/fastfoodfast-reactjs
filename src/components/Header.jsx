import React from 'react';

export default class Header extends React.Component {
  constructor() {
    super();
    this.state = {}
  }

  render() {
    return (
      <div>
        <header id="home-header">
          <div className="logo">
            <h1><a href="index.html">Fast-Food-Fast</a></h1>
          </div>
          <nav>
            <ul className="s-hide">
              <li>
                <a href="javascript:;" id="shoppingcart-b" className="shoppingcart-b"><i className="ti-shopping-cart"></i> <sup><span className="badge itemsincart">0</span></sup></a>
              </li>
              <li className="user-logout">
                <a href="profile.html"><i className="ti-user"></i></a>
              </li>
              <li className="user-logout">
                <a href="javascript:;" className="logout"><i className="ti-power-off"></i></a>
              </li>
              <li className="user-login"><a href="login.html">Login</a></li>
              <li className="user-login"><a href="signup.html">Sign Up</a></li>
            </ul>
            <div id="shoppingcart" className="cart-div hide">
              <p>You have <span className="itemsincart">0</span> item(s)</p>
              <ul className="cart-details">
                <div id="cart-details"></div>
                <li>
                  <a href="order.html"><button>View Cart</button></a>
                </li>
                <div className="clear"></div>
              </ul>
            </div>
            <ul className="l-hide">
              <li>
                <a href="javascript:;" id="shoppingcart-b" className="shoppingcart-b"><i className="ti ti-shopping-cart"></i><sup className="itemsincart">0</sup></a>
              </li>
              <li className="">
                <a href="javascript:;" className="menu" id="toggle"><i className="ti ti-user"></i></a>
                <div className="dropdown hide" id="dropdown">
                  <ul>
                    <li className="user-logout"><a href="profile.html">Profile</a></li>
                    <li className="user-logout"><a href="javascript:;" className="logout">Logout</a></li>
                    <li className="user-login"><a href="login.html">Login</a></li>
                    <li className="user-login"><a href="signup.html">Sign Up</a></li>
                  </ul>
                </div>
              </li>
            </ul>
          </nav>
        </header>
      </div>
    )
  }
}