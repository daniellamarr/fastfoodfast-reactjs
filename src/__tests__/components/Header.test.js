/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import 'babel-polyfill';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header.jsx';

const initialState = {
  cart: {
    cart: [
      {
        id: 1,
        image: 'some url',
        title: 'test title',
      },
    ],
    noOfItems: 0,
  },
};

let wrapper;

const props = {
  userData: () => {},
  getCartItems: () => {},
  auth: {
    user: {},
  },
  cart: {
    cart: [
      {
        id: 1,
        image: 'some url',
        title: 'test title',
      },
    ],
    noOfItems: 0,
  },
};

const middlewares = [thunk];

const mockStore = configureMockStore(middlewares);
const store = mockStore(initialState);

beforeEach(() => {
  wrapper = shallow(
    <Provider store={store}>
      <MemoryRouter>
        <Header {...props} />
      </MemoryRouter>
    </Provider>,
  );
});

afterEach(() => {
  wrapper.unmount();
});

describe('Header Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
