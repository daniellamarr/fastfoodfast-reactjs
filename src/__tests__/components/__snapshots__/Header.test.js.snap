// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {
        "cart": Object {
          "cart": Array [
            Object {
              "id": 1,
              "image": "some url",
              "title": "test title",
            },
          ],
          "noOfItems": 0,
        },
      },
    }
  }
>
  <MemoryRouter>
    <Header
      auth={
        Object {
          "user": Object {},
        }
      }
      cart={
        Object {
          "cart": Array [
            Object {
              "id": 1,
              "image": "some url",
              "title": "test title",
            },
          ],
          "noOfItems": 0,
        }
      }
      getCartItems={[Function]}
      userData={[Function]}
    />
  </MemoryRouter>
</ContextProvider>
`;
